envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/AsyncRestTemplateControlPlaneClient.kt

```diff
@@ -1,14 +1,26 @@
 package pl.allegro.tech.servicemesh.envoycontrol.synchronization
 
+import io.micrometer.core.instrument.MeterRegistry
 import org.springframework.web.client.AsyncRestTemplate
 import pl.allegro.tech.servicemesh.envoycontrol.services.ServicesState
 import reactor.core.publisher.Mono
 import java.net.URI
+import java.util.concurrent.TimeUnit
 
-class AsyncRestTemplateControlPlaneClient(val asyncRestTemplate: AsyncRestTemplate) : AsyncControlPlaneClient {
-    override fun getState(uri: URI): Mono<ServicesState> =
-        asyncRestTemplate.getForEntity<ServicesState>("$uri/state", ServicesState::class.java)
-            .completable()
-            .thenApply { it.body }
-            .let { Mono.fromCompletionStage(it) }
+class AsyncRestTemplateControlPlaneClient(
+    private val asyncRestTemplate: AsyncRestTemplate,
+    private val meterRegistry: MeterRegistry
+) : AsyncControlPlaneClient {
+    override fun getState(uri: URI): Mono<ServicesState> {
+        return Mono.fromCompletionStage {
+            asyncRestTemplate.getForEntity("$uri/state", ServicesState::class.java)
+                .completable()
+                .thenApply { it.body }
+        }
+            .elapsed()
+            .map { t ->
+                meterRegistry.timer("sync-dc-get-state.time").record(t.t1, TimeUnit.MILLISECONDS)
```

Comment:
Maybe create a metric segment for sync-dc. For example change `sync-dc-get-state.time` to `sync-dc.get-state.time`